tests/simplify.py

```diff
@@ -159,20 +157,44 @@ def record_node(self, input_id, is_sample=False):
         self.node_id_map[input_id] = output_id
         return output_id
 
+    def unrecord_node(self, input_id, output_id):
+        """
+        Remove the mapping for the specified input and output node pair. This is
+        done because there are no edges referring to the node.
+        """
+        assert output_id == len(self.tables.nodes) - 1
+        assert output_id == self.node_id_map[input_id]
+        self.tables.nodes.truncate(output_id)
+        self.node_id_map[input_id] = -1
+
     def flush_edges(self):
         """
         Flush the edges to the output table after sorting and squashing
         any redundant records.
         """
+        num_edges = 0
         for child in sorted(self.edge_buffer.keys()):
             for edge in self.edge_buffer[child]:
                 self.tables.edges.add_row(edge.left, edge.right, edge.parent, edge.child)
+                num_edges += 1
         self.edge_buffer.clear()
+        return num_edges
 
     def record_edge(self, left, right, parent, child):
         """
         Adds an edge to the output list.
         """
+        if self.minimise_topology:
+            # print("edge = ", left, right, parent, child)
+            X = self.edge_position_table
+            left_index = np.searchsorted(X, left)
+            right_index = np.searchsorted(X, right)
+            if left_index == right_index or (left_index == 0 and right_index == 1):
+                return
+            if left_index == 1:
+                left_index = 0
+            left = X[left_index]
+            right = X[right_index]
```

Comment:
Very nice. Just saying in words, because it took me a while to verify it's doing the right thing: This is replacing `left` and `right` with the smallest site position greater than or equal to them, i.e., slides each endpoint of an interval to the right until they hit a site position.

Reply:
I don't see what the ` or (left_index == 0 and right_index == 1)` is for, though?

Reply:
To include an option to retain trees at integer sites also, I think we could change this to:
```
        if self.minimise_topology:
            X = self.edge_position_table
            left_index = np.searchsorted(X, left)
            right_index = np.searchsorted(X, right)
            # because there are two copies of 0.0 at the start of X
            if left_index == 1:
                left_index = 0
            left_pos = X[left_index]
            right_pos = X[right_index]
            if keep_integer_trees:
                left_pos = min(left_pos, np.ceil(left))
                right_pos = min(right_pos, np.ceil(right))
            if left_pos == right_pos:
                return
            left = left_pos
            right = right_pos
```